tensorboard/backend/event_processing/event_accumulator.py

```diff
@@ -369,19 +384,17 @@ def _ProcessEvent(self, event):
     elif event.HasField('summary'):
       for value in event.summary.value:
         if value.HasField('metadata'):
+          # We only store the first instance of the metadata. This use
+          # of `setdefault` is important: the `FileWriter` does strip
+          # metadata from all values except the first one per each tag,
+          # but a new `FileWriter` is created every time a training job
+          # stops and restarts. Hence, we must also ignore non-initial
+          # metadata in this logic.
+          self.summary_metadata.setdefault(value.tag, value.metadata)
```

Comment:
Indeed, your version is slightly better not only in form but in function, as it avoids a subtle bug in the original version (that persisted in my revised version): if there are two `summary_metadata` for one tag, which contain distinct sets of plugin data, then we want to only record the plugins from the first summary, not both. Will change.